### PR TITLE
Issue#189 StatusMappings and DetailSections KAM Support initial code

### PR DESCRIPTION
**Pull request by susanodom**
https://github.com/kappnav/issues/issues/189

### CODE REVIEW

File: src/main/java/application/rest/v1/StatusEndpoint.java [hunk 1]
```diff
@@ -73,6 +73,13 @@ public Response computeStatus(@Pattern(regexp = NAME_PATTERN_ONE_OR_MORE) @PathP
             @PathParam("resource-kind") @Parameter(description = "The Kubernetes resource kind for the resource") String kind,
             @Pattern(regexp = API_VERSION_PATTERN_ZERO_OR_MORE) @DefaultValue("") @QueryParam("apiversion") @Parameter(description = "The apiVersion of the resource") String apiVersion,
             @Pattern(regexp = NAME_PATTERN_ZERO_OR_MORE) @DefaultValue("default") @QueryParam("namespace") @Parameter(description = "The namespace of the resource") String namespace) {
+        if (Logger.isEntryEnabled()) {
```
Comment (cvignola) [inline src/main/java/application/rest/v1/StatusEndpoint.java:76]: General comment:  I think we need to revisit the configmap caching.  It appears only configmaps from the kappnav ns (so-called 'global' ns) are cached.  The KAM support increases the likelihood (on purpose) that config maps come from other namespaces.  So caching should work for action/status/detail config maps that come from anywhere.

File: src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java [hunk 2]
```diff
@@ -63,58 +64,106 @@
 
     public SectionConfigMapProcessor(String kind) {
         this.sectionName= SECTION_CONFIG_MAP_NAME;   
-        this.sectionNameWithKind = SECTION_CONFIG_MAP_NAME + kind.toLowerCase(Locale.ENGLISH);
+        this.sectionNameWithKind = this.sectionName + kind.toLowerCase(Locale.ENGLISH);
```
Comment (cvignola) [inline src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java]: What is use of SECTION_CONFIG_MAP_NAME when KAM specifies name of config map?

File: src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java [hunk 3]
```diff
@@ -63,58 +64,106 @@
...
+                                    String configMapName, ConfigMapBuilder builder) {
+        if (Logger.isEntryEnabled()) {
+            Logger.log(className, "getConfigMap", Logger.LogType.ENTRY, "For namespace=" + namespace + 
+                   ", configMapName=" + configMapName);
+        }
+
         // Return the map from the local cache if it's been previously loaded.
-        if (kappnavNSMapCache.containsKey(configMapName)) {
-            return kappnavNSMapCache.get(configMapName);
+        final boolean isGlobalNS = GLOBAL_NAMESPACE.equals(namespace);
```
Comment (cvignola) [inline src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java:117]: This ties back to first comment at top of this page:  need to revisit configmap caching. Why kappnav namespace only?  I think we need to open it up to any/all namespaces that contribute a config map as per KAM mappings.

File: src/main/java/application/rest/v1/StatusEndpoint.java [hunk 4]
```diff
@@ -73,6 +73,13 @@ public Response computeStatus(@Pattern(regexp = NAME_PATTERN_ONE_OR_MORE) @PathP
...
+            Logger.log(StatusEndpoint.class.getName(), "computeStatus", Logger.LogType.ENTRY,
```
Comment (cvignola) [inline src/main/java/application/rest/v1/StatusEndpoint.java:77]: General comment:  need test cases.  Especially to validate that section and detail maps from other namespaces work and that pre-installed status/detail maps still work.

**Review by cvignola — CHANGES_REQUESTED** [review]

File: src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java (hunk 2 above)
Comment (susanodom) [reply src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java]: It was there before - I can remove it.

File: src/main/java/application/rest/v1/StatusEndpoint.java (hunk 1 above)
Comment (susanodom) [reply src/main/java/application/rest/v1/StatusEndpoint.java:76]: There are actually two configmap caches for actions and status mapping , one for "global' ns, another one is for all configmaps using ConfigMapCache class.

File: src/main/java/application/rest/v1/StatusEndpoint.java (hunk 4 above)
Comment (susanodom) [reply src/main/java/application/rest/v1/StatusEndpoint.java:77]: I do have some test cases which are needed be cleaned up and checked in.

File: src/main/java/application/rest/v1/StatusEndpoint.java (hunk 1 above)
Comment (cvignola) [reply src/main/java/application/rest/v1/StatusEndpoint.java:76]: I now recall the use of the GLOBAL_NAMESPACE constant is so the code caches config maps when the API server is running in kAppNav's namespace and does not cache with running in the actdev namespace.  There is no issue here.  Consider this resolved.

File: src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java (hunk 3 above)
Comment (cvignola) [reply src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java:117]: kappnav namespace only so it does not cache for the actdev tool, which runs in the actdev namespace.

**Review by cvignola — APPROVED** [review]

**Pull request merged**